cmake/Components.cmake

```diff
@@ -0,0 +1,146 @@
+include(CMakeParseArguments)
+
+set(global_prefix "${PROJECT_NAME}_")
+set(component_prefix "${PROJECT_NAME}_component_")
+
+define_property(GLOBAL PROPERTY "${global_prefix}INCLUDE_DIRS"
+  BRIEF_DOCS "Global include directories used by all components."
+  FULL_DOCS "Global include directories used by all components."
+)
+define_property(GLOBAL PROPERTY "${global_prefix}COMPONENTS"
+  BRIEF_DOCS "List all known Aikido components."
+  FULL_DOCS "List all known Aikido components."
+)
+
+#==============================================================================
+function(add_component component)
```

Comment:
Ok, I am officially giving up on namespacing. If you use a `macro` to generate the namespaced functions you trash all usage of `ARGN`, etc making variadic functions impossible. If you use a `function` then text substitution stops as the inner `function` definitions, and you cannot pass `${package_name}` to the aliased function (because that code isn't executed until the inner function is called, outside the scope of the defining function).

The question is do we want to wrap everything in a setup macro and do all the text substitution instead of just using `${PROJECT_NAME}`? @mkoval thoughts?